src/base/bittorrent/sha1hash.cpp

```diff
@@ -68,7 +53,24 @@ SHA1Hash::operator lt::sha1_hash() const
     return m_nativeHash;
 }
 
-SHA1Hash::operator QString() const
+SHA1Hash SHA1Hash::fromString(const QString &hashString)
+{
```

Comment:
Why not use an explicit constructor?

Reply:
I don't like constructors that convert from something that doesn't have a one-to-one correspondence with the target class. I prefer to highlight such kind of conversion.